packages/filesystem/src/electron-browser/file-dialog/electron-file-dialog-service.ts

```diff
@@ -95,6 +95,7 @@ export class ElectronFileDialogService extends DefaultFileDialogService {
         const defaultPath = FileUri.fsPath(uri);
         const filters: FileFilter[] = [{ name: 'All Files', extensions: ['*'] }];
         if (props.filters) {
+            filters.splice(0);
             filters.push(...Object.keys(props.filters).map(key => ({ name: key, extensions: props.filters![key] })));
         }
```

Comment:
@danarad05 rather than setting `all files` then removing it with `splice`, wouldn't the following be clearer as to the intentions of the method?

```suggestion
        const filters: FileFilter[] = [];
        if (props.filters) {
            filters.push(...Object.keys(props.filters).map(key => ({ name: key, extensions: props.filters![key] })));
        } else {
            filters.push({ name: 'All Files', extensions: ['*'] });
        }
```

Reply:
Fixed. I thought most cases would be without filters but fine like this also.